unblob/handlers/archive/arc.py

```diff
@@ -50,22 +50,26 @@ def calculate_chunk(
         self, file: io.BufferedIOBase, start_offset: int
     ) -> Optional[ValidChunk]:
 
-        # we loop from header to header until we reach the end header
-        offset = start_offset
-        while True:
-            file.seek(offset)
-            try:
+        try:
+            # we loop from header to header until we reach the end header
+            offset = start_offset
+            while True:
+                file.seek(offset)
                 read_bytes = file.read(2)
-            except EOFError:
-                logger.warning("Potential ARC header is missing")
-                return
-
-            if read_bytes == END_HEADER:
-                offset += 2
-                break
-            file.seek(offset)
-            header = self.parse_header(file)
-            offset += len(header) + header.size
+
+                if read_bytes == END_HEADER:
+                    offset += 2
+                    break
+                file.seek(offset)
+                header = self.parse_header(file)
+                offset += len(header) + header.size
+
+        except (EOFError, OSError):
+            # OSError triggered when seeking past the EOF
```

Comment:
Can you show an example? I believe this is not true. We just talked about this with @kissgyorgy, that `seek` does not trigger any error in this case.

His example to prove:
```python
>>> from pathlib import Path
>>> path = Path("pyproject.toml")
>>> path.stat().st_size
744
>>> fp = path.open("rb")
>>> fp.seek(1000)
1000
>>> fp.tell()
1000
>>> fp.read()
b''
```

Reply:
>       lseek() allows the file offset to be set beyond the end of the
>       file (but this does not change the size of the file).

Reply:
Finally found what triggered the OSError: negative seeking !

```
>>> with open("/tmp/garbage.bin", "rb") as f:
...     f.seek(-1)
... 
Traceback (most recent call last):
  File "<stdin>", line 2, in <module>
OSError: [Errno 22] Invalid argument
```

Reply:
Fixed this by switching the ARC header size from `long `to `ulong`. Having a negative size is not right.